src/hotspot/share/opto/loopnode.cpp

```diff
@@ -4319,13 +4319,21 @@ void PhaseIdealLoop::mark_loop_associated_parse_predicates_useful() {
   }
 }
 
+// This visitor marks all visited Parse Predicates useful.
+class ParsePredicateUsefulMarker : public PredicateVisitor {
+ public:
+  using PredicateVisitor::visit;
```

Comment:
Why is this needed?

Reply:
Forgot to explain the reason behind. Without this, compilation fails with:
```
/home/christian/jdk3/open/src/hotspot/share/opto/predicates.hpp:240:16: error: 'virtual void PredicateVisitor::visit(const InitializedAssertionPredicate&)' was hidden [-Werror=overloaded-virtual=]
  240 |   virtual void visit(const InitializedAssertionPredicate& initialized_assertion_predicate) {}
      |                ^~~~~
/home/christian/jdk3/open/src/hotspot/share/opto/loopnode.cpp:4327:8: note:   by 'virtual void ParsePredicateUsefulMarker::visit(const ParsePredicate&)'
 4327 |   void visit(const ParsePredicate& parse_predicate) override {
      |  
```
I was not quite sure what this means, so I looked it up and found this:
https://stackoverflow.com/questions/18515183/c-overloaded-virtual-function-warning-by-clang

It looks like the warning is here to prevent accidental hiding of overloads. Here is another example:
```
struct Base {
  virtual void foo(double e) {
    printf("Base");
  }
};

struct Derived: public Base {
  virtual void foo(int e) {
    printf("Derived");
  }
};

int main() {
  Derived derived;
  derived.foo(3.4f);
  return 0;
}
```
You would expect that this prints `Base`. But since the lookup happens on `Derived`, it finds a matching `Derived::foo(int e)` and applies that one (we print "Derived") even though we intended to call `Base::foo()`. By adding `using Base::foo`, we make the overloaded version available in `Derived` and pick that one instead when running `main()` (we print "Base", as anticipated).

In my example, it would not be a problem. But I guess the warning is here to just generally warn the user if the code was intended or just a typo resulting in unexpected behavior.